src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,134 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+
+
+
+/**
+ * The Storage class deals with handling the writing and reading of data to the file specified
+ */
+
+public class Storage {
+    private final File dataFile;
+    @SuppressWarnings({"checkstyle:AbbreviationAsWordInName", "CheckStyle"})
+    private final String FILE_PATH;
+
+    /**
+     * Initializes a storage object
+     *
+     * @param filePath String representing the filepath to load the file
+     */
+    public Storage(String filePath) {
+        this.FILE_PATH = filePath;
+        dataFile = new File(filePath);
+        try {
+            if (dataFile.createNewFile()) {
+                System.out.println("data file has been created");
+            }
+        } catch (IOException err) {
+            System.out.println("error opening file");
+        }
+    }
+
+    /**
+     * Appends task to the file
+     *
+     * @param task String representing task to be appended
+     */
+
+    public void appendFile (String task) {
+        try {
+            FileWriter fw = new FileWriter(FILE_PATH, true);
+            fw.write(task);
+            fw.close();
+        } catch (IOException err) {
+            System.out.println("error writing " + task + " to file storage");
+        }
+    }
+
+    /**
+     * Overwrites the file with data from the list given
+     *
+     * @param list representing the list of tasks
+     */
+    public void overwriteFile (ArrayList<Task> list) {
+        try {
+            FileWriter fw = new FileWriter(FILE_PATH);
+            for (Task currentTask : list) {
+                String done = currentTask.isDone() ? "1" : "0";
+                if (currentTask instanceof Event) {
```

Comment:
Perhaps you could make use of polymorphism here, since all 3 task types extends the `Task` class. This can help make the code a lot more succinct and readable.